common/djangoapps/student/tests/test_create_account.py

```diff
@@ -278,6 +279,16 @@ def test_discussions_email_digest_pref(self, digest_enabled):
             else:
                 self.assertIsNone(preference)
 
+    def test_referral_attribution(self):
+        affiliate_id = 'test-partner'
+        self.client.cookies['affiliate_id'] = affiliate_id
```

Comment:
Define the cookie name as a setting...just in case someone else wants to change it. No need to bother with configuration at this time, however.